### PR TITLE
IGNITE-20358 Make data storage engine configuration local

### PR DESCRIPTION
**Pull request by kgusakov**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java [hunk 1]
```diff
@@ -265,7 +265,9 @@ void nodeConfigUpdateSuggested(ParsedLine givenParsedLine) {
         // wait for lazy init of node config completer
         await("For given parsed words: " + givenParsedLine.words()).until(
                 () -> complete(givenParsedLine),
-                containsInAnyOrder("rest", "clientConnector", "network", "cluster", "deployment", "nodeAttributes")
+                containsInAnyOrder(
+                        "rest", "clientConnector", "network", "cluster", "deployment", "nodeAttributes", "aimem", "aipersist", "rocksDb"
```
Comment (ibessonov) [inline modules/cli/src/integrationTest/java/org/apache/ignite/internal/cli/core/repl/executor/ItIgnitePicocliCommandsTest.java:269]: I believe we should use lower case for "rocksdb", but let's discuss it in the future

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java [hunk 2]
```diff
@@ -778,12 +778,21 @@ private class Node {
...
-            );
+                    List.of(
+                            NetworkConfiguration.KEY,
+                            RestConfiguration.KEY,
+                            ClientConnectorConfiguration.KEY,
+                            PersistentPageMemoryStorageEngineConfiguration.KEY,
+                            VolatilePageMemoryStorageEngineConfiguration.KEY),
+                    List.of(),
+                    List.of(UnsafeMemoryAllocatorConfigurationSchema.class)
+                    );
```
Comment (ibessonov) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceDistributedTest.java:789]:
```suggestion
            );
```

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/configuration/PageMemoryStorageDistributedConfigurationModule.java [hunk 3]
```diff
@@ -44,7 +42,7 @@ public ConfigurationType type() {
     /** {@inheritDoc} */
     @Override
     public Collection<RootKey<?, ?>> rootKeys() {
-        return List.of(VolatilePageMemoryStorageEngineConfiguration.KEY, PersistentPageMemoryStorageEngineConfiguration.KEY);
+        return List.of();
```
Comment (ibessonov) [inline modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/configuration/PageMemoryStorageDistributedConfigurationModule.java:45]: You forgot about other methods. Why did you left the class if there's no distributed storage configuration anymore? You could have renamed it at that's it.

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/configuration/RocksDbStorageDistributedConfigurationModule.java [hunk 4]
```diff
@@ -42,7 +41,7 @@ public ConfigurationType type() {
     /** {@inheritDoc} */
     @Override
     public Collection<RootKey<?, ?>> rootKeys() {
-        return List.of(RocksDbStorageEngineConfiguration.KEY);
+        return List.of();
```
Comment (ibessonov) [inline modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/configuration/RocksDbStorageDistributedConfigurationModule.java:44]: Same here

**Review by ibessonov — CHANGES_REQUESTED** [review]

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/configuration/PageMemoryStorageDistributedConfigurationModule.java (hunk 3 above)
Comment (kgusakov) [reply modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/configuration/PageMemoryStorageDistributedConfigurationModule.java]: The whole class is removed

File: modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/configuration/RocksDbStorageDistributedConfigurationModule.java (hunk 4 above)
Comment (kgusakov) [reply modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/configuration/RocksDbStorageDistributedConfigurationModule.java]: Removed

File: modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/configuration/PageMemoryStorageEngineLocalConfigurationModule.java [hunk 5]
```diff
@@ -52,8 +52,7 @@ public ConfigurationType type() {
     public Collection<Class<?>> polymorphicSchemaExtensions() {
         return List.of(
                 VolatilePageMemoryDataStorageConfigurationSchema.class,
-                PersistentPageMemoryDataStorageConfigurationSchema.class
-        );
+                PersistentPageMemoryDataStorageConfigurationSchema.class);
```
Comment (ibessonov) [inline modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/configuration/PageMemoryStorageEngineLocalConfigurationModule.java:55]: Useless reformatting :)
Alright, I'll allow it this time

**Review by ibessonov — APPROVED** [review]

**Pull request merged**